core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -62,14 +70,56 @@ class BaseSnapshot implements Snapshot {
       Map<String, String> summary,
       Integer schemaId,
       String manifestList) {
+    this(
+        sequenceNumber,
+        snapshotId,
+        parentId,
+        timestampMillis,
+        operation,
+        summary,
+        schemaId,
+        manifestList,
+        0L,
+        null,
+        null,
+        null);
+  }
+
+  BaseSnapshot(
+      long sequenceNumber,
+      long snapshotId,
+      Long parentId,
+      long timestampMillis,
+      String operation,
+      Map<String, String> summary,
+      Integer schemaId,
+      String manifestListLocation,
+      long manifestListSize,
+      ByteBuffer manifestListKeyMetadata,
+      ByteBuffer wrappedManifestListKeyMetadata,
+      String wrappedKeyEncryptionKey) {
     this.sequenceNumber = sequenceNumber;
     this.snapshotId = snapshotId;
     this.parentId = parentId;
     this.timestampMillis = timestampMillis;
     this.operation = operation;
     this.summary = summary;
     this.schemaId = schemaId;
-    this.manifestListLocation = manifestList;
+    this.manifestListLocation = manifestListLocation;
+    this.manifestListSize = manifestListSize;
+    this.manifestListKeyMetadata = manifestListKeyMetadata;
+    if (wrappedManifestListKeyMetadata != null) {
+      Preconditions.checkArgument(
+          wrappedManifestListKeyMetadata.arrayOffset() == 0, "Offset must be 0");
```

Comment:
I appreciate that this is checking the offset rather than ignoring it, but it is always better to use utils in `ByteBuffers` instead. What about using `ByteBuffers.toByteArray` for this?

Reply:
Looks like there are several places that call `ByteBuffer.array` directly. Can you replace all of them with `ByteBuffers.toByteArray`?